src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -286,6 +286,21 @@ theorem getLsbD_ofNat (n : Nat) (x : Nat) (i : Nat) :
 
 @[simp] theorem getMsbD_zero : (0#w).getMsbD i = false := by simp [getMsbD]
 
+@[simp] theorem getLsbD_one : (1#w).getLsbD i = (decide (0 < w) && decide (0 = i)) := by
+  simp only [getLsbD, toNat_ofNat, Nat.testBit_mod_two_pow]
+  by_cases h : i = 0
+  <;> simp [h, Nat.testBit_to_div_mod, Nat.div_eq_of_lt]
+  omega
+
+@[simp] theorem getElem_one (h : i < w) : (1#w)[i] = decide (0 = i) := by
+  simp [← getLsbD_eq_getElem, getLsbD_one, h, show 0 < w by omega]
+
+@[simp] theorem getMsbD_one : (1#w).getMsbD i = decide (0 = w - 1 - i ∧ i < w) := by
+  simp only [getMsbD]
+  by_cases h : i < w
+  · simp [h, show 0 < w by omega]
+  · simp [h]
```

Comment:
```suggestion
/-- The msb at index `w-1` is the least significant bit, and is true when the width is nonzero.
@[simp] theorem getMsbD_one : (1#w).getMsbD i = decide (i = w - 1 ∧ 0 < w) := by
  simp only [getMsbD]
  by_cases h : 0 < w <;> by_cases h' : i = w - 1 <;> simp [h, h'] <;> omega
```

Let's write the RHS such that get exploitable hypotheses, where we show that `i = w - 1` and that the width is positive. It's also, at least to me, a little more conceptual.